File: examples/typescript/sdk-reference/serialization/filesystem-serdes-overflow.ts
import {
  createFileSystemSerdes,
  FileSystemSerdesMode,
} from "@aws/durable-execution-sdk-js";

// Small payloads stay inline in the checkpoint. The SDK only writes to the
// filesystem when the checkpoint envelope would exceed the size limit.
export const overflowFileSystemSerdes = createFileSystemSerdes("/mnt/s3", {
  storageMode: FileSystemSerdesMode.OVERFLOW,
});

File: examples/python/sdk-reference/serialization/filesystem-serdes-overflow.py
from aws_durable_execution_sdk_python.filesystem_serdes import (
    FileSystemSerDes,
    FileSystemSerDesConfig,
    FileSystemSerDesMode,
)

# In OVERFLOW mode, small payloads stay inline in the checkpoint and no file
# is written. The SDK only writes to the filesystem when the serialized value
# exceeds the checkpoint size limit.
overflow_fs_serdes = FileSystemSerDes(
    "/mnt/s3",
    FileSystemSerDesConfig(storage_mode=FileSystemSerDesMode.OVERFLOW),
)
